### ISSUE TITLE
Missing headers in plugins implementing [K]SNI

### ISSUE THREAD

**Issue by pmattern** [lxqt/lxqt-connman-applet#10]
The headers in LXQt ConnMan applet's menu aren't displayed if the applet is handled by a panel component implementing [K]SNI.

Seen running 7af99fc on Arch Linux i686 or x86_64: with LXQt and LXQt ConnMan applet from recent VCS checkouts the headers are displayed as usual in plugin-tray implementing System Tray Protocol but aren't in plugin-statusnotifier implementing SNI
![lxqt-connman-applet_missing-menu-headers_lxqt](https://cloud.githubusercontent.com/assets/9860801/10085722/0284c25c-630b-11e5-855f-4242dbe36bd0.png)
Same findings in "System Tray" of KDE Plasma 5.4.1.
![lxqt-connman-applet_missing-menu-headers_kde](https://cloud.githubusercontent.com/assets/9860801/10085728/0a677456-630b-11e5-88e7-181af86655b1.png)
To check the latter libqtxdg, liblxqt and LXQt ConnMan applet were compiled on a system lacking all other LXQt components.

**Comment by surlykke** [comment]
Hmm. This is tricky.
- Adding a custom widget (which is what was done in lxqt-connman-applet)
  doesn't work.
- Calling addSection(_text_) doesn't work.
- Adding a simple menu entry does work (obviously), but the font can't be
  changed (made bold).

I rather suspect it's a qt-bug - are there other qt-based applications
displaying in systemtray/sni that shows section headings?
- Chr.

2015-09-24 22:31 GMT+02:00 Peter Mattern notifications@github.com:

**Comment by pmattern** [comment]
> are there other qt-based applications displaying in systemtray/sni that shows section headings?

Not that I knew of, sorry. Will report back at once if I should stumble upon one.

Maybe @paulolieuthier can drop a note here? He knows that whole SNI thing very well.

If it should turn out that headers like used by lxqt-connman-applet cannot be realized in SNI, would making the applet a regular LXQt panel plugin actually be an option, too?

**Comment by paulolieuthier** [comment]
Yeah, that's because the menus are passed to the status notifier through DBus, so style data is not kept. [Here](http://bazaar.launchpad.net/~torkvemada/libdbusmenu-qt/libdbusmenu-qt/view/head:/src/com.canonical.dbusmenu.xml) is the xml interface. You can use the primary button click to open your custom menu.

**Comment by surlykke** [comment]
Aha, ok.

Thanks for educating me :-)

I'll switch to using the primary button.

br. Chr.

2015-09-27 4:47 GMT+02:00 Paulo Lieuthier notifications@github.com:

**Comment by surlykke** [comment]
Ok.
I've decided for a rewrite, moving functionality from the context menu to a widget as the somewhat limited interface of SNI does not allow any type of rich menu. This should fix this issue.

**Comment by pmattern** [comment]
0431468 segfaults right upon launch on Arch Linux x86_64 running LXQt from yesterday's VCS checkouts here.

Problem can be seen no matter whether only one of plugin-statusnotifier and plugin-tray, both or none is installed.
Regression testing suggests it was introduced in 7f8a030.
Nothing helpful in systemd's journal, backtrace is

```
#0  0x00007f2d39264cc0 in QString::startsWith(QChar, Qt::CaseSensitivity) const () from /usr/lib/libQt5Core.so.5
#1  0x00007f2d3a8626b2 in QDBusUtil::isValidObjectPath(QString const&) () from /usr/lib/libQt5DBus.so.5
#2  0x00007f2d3a8554b1 in ?? () from /usr/lib/libQt5DBus.so.5
#3  0x00007f2d3a857aec in QDBusAbstractInterface::QDBusAbstractInterface(QString const&, QString const&, char const*, QDBusConnection const&, QObject*) () from /usr/lib/libQt5DBus.so.5
#4  0x0000000000434666 in NetConnmanTechnologyInterface::NetConnmanTechnologyInterface(QString const&, QString const&, QDBusConnection const&, QObject*) ()
#5  0x00000000004148da in TechnologyFrame::TechnologyFrame(QString, QMap<QString, QVariant> const&, QWidget*) ()
#6  0x000000000041acb9 in ServicesWindow::onTechnologyAdded(QDBusObjectPath const&, QMap<QString, QVariant> const&) ()
#7  0x000000000041b45e in ServicesWindow::ServicesWindow(QWidget*) ()
#8  0x0000000000413505 in main ()
```

**Comment by surlykke** [comment]
Can you check if 440c3d31ca9d69 helps?
- Chr.

2015-10-10 19:11 GMT+02:00 Peter Mattern notifications@github.com:

**Comment by pmattern** [comment]
At first glance the problem seems to be fixed in 440c3d3. I'll keep an eye on this and report back if needed.

**Comment by surlykke** [comment]
Thanks. And thanks for reporting.
- Chr.

2015-10-10 23:09 GMT+02:00 Peter Mattern notifications@github.com:
